File: hook.py
import struct

f = open("/dev/input/event2", "rb")
counter = 0
while 1:
  data = f.read(24)
  unpack = struct.unpack('4IHHI', data)
  counter+=1
  if(counter==2 and int(unpack[6])==1):
    print(unpack[5])
  if(counter==3):
    counter=0

File: hook.cpp
#include <fcntl.h>
#include <linux/input.h>
#include <stdio.h>
#include <stdlib.h>
#include <unistd.h>

int main() {
    int fd = open("/dev/input/event2", O_RDONLY);
    if (fd == -1) {
        perror("open");
        exit(EXIT_FAILURE);
    }

    int counter = 0;
    while (true) {
        input_event event;
        if (read(fd, &event, sizeof(event)) != sizeof(event)) {
            perror("read");
            exit(EXIT_FAILURE);
        }

        counter++;
        if (counter == 2 && event.type == EV_KEY && event.value == 1) {
            printf("%d\n", event.code);
        }
        if (counter == 3) {
            counter = 0;
        }
    }

    close(fd);
    return 0;
}